### ISSUE TITLE
[macOS] Failed to load credential plugin

### ISSUE THREAD

**Issue by LermanR** [njust/kubelog#3]
When opening file (ctrl+click -> open)
I get the error message "Failed to load namespaces"

I can see the clusters list on the left, but I get the same error in all of them.

macOS Big Sur
Clusters on aws eks
when running "kubectl get namespace" -  I get the list

**Comment by paulofrazao** [comment]
Same here, just installed and getting the same error.

**Comment by njust** [comment]
Hi, thanks for reporting this. Which Kubernetes version do you have on your cluster?

**Comment by paulofrazao** [comment]
v1.26

**Comment by LermanR** [comment]
v1.21

**Comment by njust** [comment]
Hi, thanks for your feedback. Please try version v0.3.3 and check the logs here: $HOME/.local/share/kubelog/logs/.

**Comment by paulofrazao** [comment]
same error and there is no log files.

**Comment by njust** [comment]
Does at least the directory exist? I'm using [data_dir](https://docs.rs/directories/5.0.0/directories/struct.ProjectDirs.html#method.data_dir) to determine the logs directory. Unfortunately, I have no mac to check, but the logs were written to the expected location  on Linux and Windows.

**Comment by njust** [comment]
Sorry, the path was for Linux systems. The correct path for macOS is: "$HOME/Library/Application\ Support/kubelog/logs/"

**Comment by LermanR** [comment]
from the log:
[2m2023-04-13T06:37:39.020212Z[0m [31mERROR[0m ThreadId(01) [2mkubelog::namespace_list_view[0m[2m:[0m Failed to load namespaces: auth error: unable to run auth exec: No such file or directory (os error 2)

**Comment by njust** [comment]
The problem is that the kube-client library cannot find your credential plugin. The relevant code is here:
https://github.com/kube-rs/kube/blob/33ba90e0a003c915801bb9b6c4961b7d0b721889/kube-client/src/client/auth/mod.rs#L542

I have not been working with credential plugins yet, but I think you have these options:
- Make sure your credential plugin is in path
- Make the path of your credential plugin in your kube-config full qualified

**Comment by LermanR** [comment]
Not sure what you mean by "credential plugin".
The credentials are in the .kube/config file. 
I've added the .kube folder to the PATH, but its still doesn't work...

**Comment by njust** [comment]
The error message you are facing seems to be a part of the [kube-rs](https://github.com/kube-rs/kube) client library trying to call a credential plugin, which is probably defined in your kube-config below the user > exec > command section e.g: 

https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuration

```yaml
apiVersion: v1
kind: Config
users:
- name: my-user
  user:
    exec:
      # Command to execute. Required.
      command: "example-client-go-exec-plugin"
      # Arguments to pass when executing the plugin. Optional.
      args:
      - "arg1"
      - "arg2"
```
If you have a credential plugin defined, then you could try:
- Add the folder of that plugin to your path
- Make that command full qualified
- Place kubelog next to your credential plugin

**Comment by LermanR** [comment]
If i understood you correctly
The command is: aws
"aws" folder is in usr/local/aws-cli

I've tried to add the folder to PATH.
Tried to place kubelog next to that folder.
And tried to assemble full command with the args as the command
command: aws --region x eks --cluster-name y

Nothing worked...

**Comment by njust** [comment]
I'm not a macOS user, but setting an environment variable which works also for a GUI app seems to be not so straight forward on macOS, and works different, depending on which macOS version you use:

https://serverfault.com/questions/16355/how-do-i-set-the-global-path-environment-variable-on-os-x
https://stackoverflow.com/questions/603785/environment-variables-in-mac-os-x/4567308#4567308

So when I wrote "Make that command full qualified" I meant that you should set the full path of the aws credential helper in the kube config.

I created a Kubernetes cluster on AWS EKS and kubelog successfully authenticated via the AWS credentials plugin, after changing the path to be full qualified. The relevant part of my kube config looks like this: 

```yaml
- name: arn:aws:eks:eu-central-1:435496144405:cluster/test
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - --region
      - eu-central-1
      - eks
      - get-token
      - --cluster-name
      - test
      - --output
      - json
      command: /usr/local/aws-cli/aws
```

**Comment by njust** [comment]
Should be fixed with v0.3.4. Feel free to reopen if something is missing.
